### PR TITLE
[ABW-3531, ABW-3544] Create & Edit persona updates

### PR DESCRIPTION
**Pull request by danvleju-rdx**
Jira ticket: [ABW-3531](https://radixdlt.atlassian.net/browse/ABW-3531)
Jira ticket: [ABW-3544](https://radixdlt.atlassian.net/browse/ABW-3544)

## Description
- Updated Create & Edit persona UI
- Removed the old Create screen and unified Create & Edit into a single screen
- Added data fields that was missing on Create screen
- Updated `AppTextField`

## Video

https://github.com/user-attachments/assets/b695fbb5-4d14-4bd8-94ea-f4f5cb890ad9

[ABW-3531]: https://radixdlt.atlassian.net/browse/ABW-3531?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ABW-3544]: https://radixdlt.atlassian.net/browse/ABW-3544?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: RadixWallet/Features/EditPersonaFeature/EditPersona/EditPersona.swift [hunk 1]
```diff
@@ -57,16 +57,16 @@ extension EditPersona {
 public struct EditPersona: Sendable, FeatureReducer {
```
Comment (danvleju-rdx) [inline RadixWallet/Features/EditPersonaFeature/EditPersona/EditPersona.swift:57]: Didn’t find a better naming that will suit both Create and Edit flows. Open to suggestions.

File: RadixWallet/MIGRATE_TO_SARGON/Stage1/Stage1MigrateToSargon+Persona.swift [hunk 2]
```diff
@@ -49,3 +49,7 @@ extension CollectionOfEmailAddresses: PersonaDataCollectionProtocol {
 extension PersonaDataIdentifiedEmailAddress: PersonaDataCollectionElement {
 	public typealias Value = PersonaDataEntryEmailAddress
 }
+
+extension Persona {
+	public static let nameMaxLength = 30
+}
```
Comment (danvleju-rdx) [inline RadixWallet/MIGRATE_TO_SARGON/Stage1/Stage1MigrateToSargon+Persona.swift:55]: Placed this constant here for now. Could move to Sargon if needed.

**Review by matiasbzurovski — CHANGES_REQUESTED** [review]
Looking great! Only comment is about the changes to `Hint`, we can discuss that on Slack if you want

File: RadixWallet/Core/DesignSystem/Components/Hint.swift [hunk 3]
```diff
@@ -60,11 +60,11 @@ public struct Hint: View, Equatable {
 						.renderingMode(.template)
 						.resizable()
 						.scaledToFit()
-						.frame(.smallest)
+						.frame(.icon)
 				}
 				text
 					.lineSpacing(0)
-					.textStyle(.body2HighImportance)
+					.textStyle(.body2Regular)
```
[the remark this answers is not in the document]
Comment (danvleju-rdx) [reply RadixWallet/Core/DesignSystem/Components/Hint.swift]: reverted

**Review by matiasbzurovski — APPROVED** [review]

File: RadixWallet/Core/DesignSystem/Components/AppTextField.swift [hunk 4]
```diff
@@ -99,30 +99,32 @@ public struct AppTextField<FocusValue: Hashable, Accessory: View, InnerAccessory
...
-								.textStyle(primaryHeading.isProminent ? .body1HighImportance : .body2Regular)
-								.foregroundColor(primaryHeading.isProminent && isEnabled ? accentColor(isFocused: true) : .app.gray2)
-								.multilineTextAlignment(.leading)
-							if let subHeading {
-								Text(subHeading)
-									.textStyle(.body2Regular)
-									.foregroundColor(.app.gray2)
-									.multilineTextAlignment(.trailing)
+			VStack(alignment: .leading, spacing: .small1) {
+				if primaryHeading != nil || secondaryHeading != nil {
```
Comment (kugel3) [inline RadixWallet/Core/DesignSystem/Components/AppTextField.swift:103]: Is this correct? The whole block will be visible if secondaryHeading is non-nil, but it looks like the block will be empty anyway, unless primaryHeading is also non-nil?

File: RadixWallet/Features/CreatePersona/Children/Introduction/IntroductionToPersonas+View.swift [hunk 5]
```diff
@@ -18,35 +12,33 @@ extension IntroductionToPersonas {
...
 				}
-				.onAppear { viewStore.send(.appeared) }
+				.onAppear { store.send(.view(.appeared)) }
 				.destinations(with: store)
 			}
 		}
 
 		@ViewBuilder
-		private func introToPersona(with viewStore: ViewStoreOf<IntroductionToPersonas>) -> some SwiftUI.View {
+		private var introToPersona: some SwiftUI.View {
```
Comment (kugel3) [inline RadixWallet/Features/CreatePersona/Children/Introduction/IntroductionToPersonas+View.swift:36]: I would just put this stuff inline, or if you really want to extract it, bring the VStack along. ViewBuilder variables with multiple elements can cause surprises later.

File: RadixWallet/Features/CreatePersona/Children/Introduction/IntroductionToPersonas+View.swift [hunk 6]
```diff
@@ -18,35 +12,33 @@ extension IntroductionToPersonas {
...
 			Image(asset: AssetResource.persona)
+				.resizable()
+				.frame(.veryHuge)
 
-			Text(viewStore.titleText)
+			Text(L10n.CreatePersona.Introduction.title)
```
Comment (kugel3) [inline RadixWallet/Features/CreatePersona/Children/Introduction/IntroductionToPersonas+View.swift:41]: This is better, strange that this used to be a variable. Perhaps at some point there were multiple options.

**Review by kugel3 — APPROVED** [review]
Good job. Also add `@Reducer` to the reducer.

File: RadixWallet/Core/DesignSystem/Components/AppTextField.swift (hunk 4 above)
Comment (kugel3) [reply RadixWallet/Core/DesignSystem/Components/AppTextField.swift:103]: Ah, I misread the indentation, I thought that was outside of this `if` block.

**Pull request merged**